plat/arm/board/fvp/aarch64/fvp_helpers.S

```diff
@@ -205,9 +203,9 @@ func plat_arm_calc_core_pos
 	ubfx	x2, x3, #MPIDR_AFF2_SHIFT, #MPIDR_AFFINITY_BITS
 
 	/* Compute linear position */
-	mov	x4, #FVP_MAX_PE_PER_CPU
-	madd	x0, x1, x4, x0
-	mov	x5, #FVP_MAX_CPUS_PER_CLUSTER
-	madd	x0, x2, x5, x0
+	mov	x3, #FVP_MAX_CPUS_PER_CLUSTER
+	madd	x1, x2, x3, x1
+	mov	x3, #FVP_MAX_PE_PER_CPU
```

Comment:
Suggest use `x4` here so as to not to depend on the write to `x3` above.

Reply:
Change them. Now, Both mov instructions are independent and may be executed out of order.